src/main/groovy/edu/oregonstate/mist/locations/db/DiningDAO.groovy

```diff
@@ -0,0 +1,164 @@
+     * it reads it from the cache.
+     *
+     * @return String json format of dining locations
+     */
+    private String getDiningLocationList() throws Exception{
+        locationUtil.getDataFromUrlOrCache(uhdsURL, diningJsonOut)
+    }
+
+    private HashMap<Integer, List<DayOpenHours>> parseDiningICAL(String icalData) {
+        // setup ical4j calendar and parse it
+        CalendarBuilder builder = new CalendarBuilder()
+        def stream = new ByteArrayInputStream(icalData.getBytes())
+        net.fortuna.ical4j.model.Calendar calendar = builder.build(stream)
+
+        // setup jodatime varaibles
+        Map<Integer, List<DayOpenHours>> weekOpenHours = new HashMap<Integer, List<DayOpenHours>>()
+        DateTime today = new DateTime().withTimeAtStartOfDay()
+
+        (0..6).each { // iterate over a week to find out next 7 days of open hours
+            def singleDay = today.plusDays(it)
+            def dayOpenHours = new ArrayList<DayOpenHours>()
+
+            // filter out so that only events for the current day are retrieved
+            def ical4jToday = new net.fortuna.ical4j.model.DateTime(singleDay.toDate())
+            Period period = new Period(ical4jToday, new Dur(1, 0, 0, 0))
+            Filter filter = new Filter(new PeriodRule(period))
+            List eventsToday = filter.filter(calendar.getComponents(Component.VEVENT))
+
+            eventsToday.each { // put break right here
+                if (!isEventExcluded(it)) { // today was excluded from event recursive rule
+                    addEventToToday(it, dayOpenHours)
+                }
+            } // iterate over today's events
+
+            weekOpenHours.put(singleDay.dayOfWeek, dayOpenHours)
+        } // iterate over weekday
+        weekOpenHours
+    }
+
+    /**
+     * Checks to see if the event's recurrence was excluded using EXDATE.
+     *
+     * @param it
+     * @return
+     */
+    private boolean isEventExcluded(def it) {
+        PropertyList exDates = it.getProperties(Property.EXDATE)
+        exDates.each { ex ->
+            ((ExDate) ex).dates.each { oneExDate ->
+                if (new DateTime(oneExDate).toLocalDate().equals(new LocalDate())) {
+                    return true
+                }
+            }
+        }
+
+        false
+    }
+
+    private void addEventToToday(def it, ArrayList<DayOpenHours> dayOpenHours) {
+        def dtStart = it.getProperties().getProperty(Property.DTSTART)
+        def dtEnd = it.getProperties().getProperty(Property.DTEND)
+        def sequence = it.getProperties().getProperty(Property.SEQUENCE)
+        def uid = it.getProperties().getProperty(Property.UID)
+
+        // Json annotation in POGO handles utc storage
+        DayOpenHours eventHours = new DayOpenHours([start: dtStart.date, end: dtEnd.date, sequence: sequence.sequenceNo,
+                                             uid  : uid.value])
```

Comment:
consider indenting map for readability. note that the square brackets are optional.